src/components/Detail.tsx

```diff
@@ -104,6 +104,21 @@ export default class Detail extends React.Component<Props, State> {
     }
   }
 
+  /**
+   * Joins values of specified field in an array of objects by the given character
+   * as long as the values are strings and truthy.
+   *
+   * @param arr array of objects
+   * @param field name of the field to look for values
+   * @param char character to use for joining (space added by default)
+   * @returns <div> element that contains values separated by a character
+   */
+  static joinFieldValuesByCharacter(arr: Array<{[key:string]: any}>, field: string, char: string) {
+    return (
+      <div>{arr.filter(obj =>  obj[field] && typeof obj[field] === 'string').map(obj => obj[field]).join(`${char} `)}</div>
```

Comment:
`arr.filter(obj =>  obj[field] && typeof obj[field] === 'string').map(obj => obj[field])` is needlessly complex when both the object and a function to extract the value from the object could be provided, e.g.

```javascript
static joinFieldValuesByCharacter<T>(arr: Array<T>, extractor: (object: T) => string, separator: string) {
  return (
    <div>{arr.map(element => extractor(element)).join(separator)}</div>
  )
}
```

The calling code:

```javascript
Detail.joinFieldValuesByCharacter(item.studyAreaCountries, c => c.country, ", ")
```

This also has the advantage of being type safe as well as being more flexible if the extractor is non-trivial.

The final `join()` inserts a space regardless of the provided separator, which may be surprising.

Reply:
Thank you for this! I made these changes but I kept filter as a part of it to deal with "" (and " ") just in case.